File: problems/hard/bear_and_gene/__tests__/test_bear_and_gene.py
import unittest
import sys
import os

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from bear_and_gene import steadyGene

class TestSteadyGene(unittest.TestCase):

    def test_example1(self):
        gene = "GAAATAAA"
        self.assertEqual(steadyGene(gene), 5)

    def test_already_steady(self):
        gene = "GACT"
        self.assertEqual(steadyGene(gene), 0)

    def test_large_balanced(self):
        gene = "AAAACCCCGGGGTTTT"
        self.assertEqual(steadyGene(gene), 0)

    def test_another_case(self):
        gene = "AAGTGCCT"
        self.assertEqual(steadyGene(gene), 0)

    def test_edge_case(self):
        gene = "AAAAACCCGGTT"
        self.assertEqual(steadyGene(gene), 2)

if __name__ == '__main__':
    unittest.main()

File: problems/hard/bear_and_gene/__tests__/test_bear_and_gene.cpp
#include <cassert>
#include <iostream>
#include "../bear_and_gene.cpp"
using namespace std;

void test() {
    assert(steadyGene("GAAATAAA") == 5);
    assert(steadyGene("GACT") == 0);
    assert(steadyGene("AAAACCCCGGGGTTTT") == 0);
    assert(steadyGene("AAGTGCCT") == 0);
    assert(steadyGene("AAAAACCCGGTT") == 2);
    cout << "All test cases passed!" << endl;
}

int main() {
    test();
    return 0;
}
